pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -174,26 +172,32 @@ func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Inter
 		}),
 		DeleteFunc: nodeutil.CreateDeleteNodeHandler(ca.ReleaseCIDR),
 	})
-	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: nodeutil.CreateAddNodeHandler(func(node *v1.Node) error {
-			if ca.nodeTopologyQueue != nil {
+
+	enableNodeTopology = enableMultiSubnetCluster
+	if enableNodeTopology {
+		nodeTopologySyncer := &NodeTopologySyncer{
+			nodeTopologyClient: nodeTopologyClient,
+			cloud:              gceCloud,
+			nodeLister:         nodeInformer.Lister(),
+		}
+		nodetopologyQueue := NewTaskQueue("nodetopologgTaskQueue", "nodetopologyCRD", nodeTopologyWorkers, nodeTopologyKeyFun, nodeTopologySyncer.sync)
+		ca.nodeTopologyQueue = nodetopologyQueue
+
+		nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: nodeutil.CreateAddNodeHandler(func(node *v1.Node) error {
 				ca.nodeTopologyQueue.Enqueue(node)
-			}
-			return nil
-		}),
-		UpdateFunc: nodeutil.CreateUpdateNodeHandler(func(oldNode, newNode *v1.Node) error {
-			if ca.nodeTopologyQueue != nil {
+				return nil
+			}),
+			UpdateFunc: nodeutil.CreateUpdateNodeHandler(func(oldNode, newNode *v1.Node) error {
```

Comment:
This need not be addressed in this PR, but one thing I should have mentioned earlier is that this Update may be a bit noisy, since there are numerous updates to Nodes and in clusters with large nodes, this can start becoming bothersome. You may want to compare the oldNode and the newNode and perhaps only Enqueue if you notice a change on one of the required labels.

Reply:
yea good call. let's do a follow up fix after this change. 

Change should be minor as I was experimenting but I wanted to figure out a proper test.